truss-chains/truss_chains/deployment/code_gen.py

```diff
@@ -704,40 +712,55 @@ def _write_truss_config_yaml(
     config.resources.use_gpu = bool(compute.accelerator.count)
     config.runtime.predict_concurrency = compute.predict_concurrency
     config.runtime.is_websocket_endpoint = is_websocket_endpoint
-    # Image.
-    _inplace_fill_base_image(chains_config.docker_image, config)
-    pip_requirements = _make_requirements(chains_config.docker_image)
-    # TODO: `pip_requirements` will add server requirements which give version
-    #  conflicts. Check if that's still the case after relaxing versions.
-    # config.requirements = pip_requirements
-    pip_requirements_file_path = chainlet_dir / _REQUIREMENTS_FILENAME
-    pip_requirements_file_path.write_text("\n".join(pip_requirements))
-    # Absolute paths don't work with remote build.
-    config.requirements_file = _REQUIREMENTS_FILENAME
-    config.system_packages = chains_config.docker_image.apt_requirements
-    if chains_config.docker_image.external_package_dirs:
-        for ext_dir in chains_config.docker_image.external_package_dirs:
-            config.external_package_dirs.append(ext_dir.abs_path)
-    config.use_local_chains_src = use_local_chains_src
+
     # Assets.
     assets = chains_config.get_asset_spec()
     config.secrets = assets.secrets
-    if definitions.BASETEN_API_SECRET_NAME not in config.secrets:
-        config.secrets[definitions.BASETEN_API_SECRET_NAME] = definitions.SECRET_DUMMY
+
+    if engine_builder_config:
+        config.trt_llm = engine_builder_config
```

Comment:
nit: could we use an early return here to avoid the additional nesting for the rest of this big block? I do see the one common line at the end though, maybe that's worth duplicating - up to you!